lib/preload.js

```diff
@@ -437,16 +434,19 @@ class Preloader extends EventEmitter {
       // Pull the image
       return innerDockerProgress.pull(appData.imageId, (event) => {
         // Emit progress events while pulling
-        percentage = event.percentage
         this._progress(pullingProgressName, event.percentage)
       })
     })
     .then(() => {
-      if (percentage !== 100) {
-        return
-      }
       // Signal that we're done to the Python script.
       this.stdin.write('\n')
+      // Wait for the script to unmount the data partition
+      return new Promise((resolve, reject) => {
+        this.stdout.once('error', reject)
```

Comment:
What errors are you expecting to handle here? Was this meant to be `this.stderr.once('data', reject)` or is this meant to handle cases where the child process errors/exits, or is this something docker.modem does (emitting errors on the stream)?

Reply:
PS: Sorry I'm a bit late to the party

Reply:
I'm not expecting an error here, I've just put it there so if an error happens it will be easier to debug.
`data` on `stderr` is expected, we use it as a debug output.

Reply:
Cool, thanks! :)